jupyter_server/base/zmqhandlers.py

```diff
@@ -259,10 +259,15 @@ def pre_get(self):
         the websocket finishes completing.
         """
         # authenticate the request before opening the websocket
-        if self.get_current_user() is None:
+        user = self.get_current_user()
+        if user is None:
             self.log.warning("Couldn't authenticate WebSocket connection")
             raise web.HTTPError(403)
```

Comment:
TBH, I would expect this to be a 401...

Reply:
Now called Authorizer

Reply:
I'm still wondering if this should be 401? I.e. no auth = 401; has auth, not permissions = 403 ?

Reply:
(also the comment didn't seem to match the topic)

Reply:
Yeah, weird - that was for another thread. I think this should be 403.

401 responses [MUST](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/401) come with a WWW-Authenticate header explaining how to authenticate with appropriate permissions. We don't have that, so we shouldn't use 401.

In practice, I think 401 is ~only used for websites that use Basic auth, which is what prompts browsers to display the old-timey login dialog.

Reply:
I think you are right (https://httpwg.org/specs/rfc7231.html#status.403 is also a good resource). 401 might also make sense for Bearer token systems, but I agree with the observation that it is mostly used with Basic 👍  Thanks for clarifying!